tests/functional/test_args_resolution.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from awscli.testutils import create_clidriver
+from awscli.testutils import BaseAWSCommandParamsTest
+
+
+class TestArgsResolution(BaseAWSCommandParamsTest):
+
+    def test_profile_resolution_order(self):
+        self.environ['AWS_PROFILE'] = 'foo'
+        self.run_cmd('--profile bar ec2 describe-instances', 255)
```

Comment:
It would be nice to have the expected rc be zero by creating a profile that exists and mocking out the response. For v1, I think a non-existent profile results in a rc of `1` , but it would be a much stronger assertion/safety net to have a test that asserts we can successfully make a CLI call where the profile overrides the environment variable that has a non-existent profile.